Docs/UPDATING.md

```diff
@@ -0,0 +1,59 @@
+# How to support new version of Xcode
+
+When Xcode updates, it might change the format of its messages to XCBBuildService. This requires changes to the proxy to support.
+
+There are two ways to debug: reading logs, and debugging using Xcode. Both have their own benefits and drawbacks, and it's likely you'll lean on both when working to update the library.
+
+## Logs
+
+This project makes thorough use of [swift-logging](https://github.com/apple/swift-log), so to see what's wrong, look in the logs!
+
+You can start recording logs by executing `write_shim.sh`. You'll need your built `BazelXCBBuildService` and the path to where Xcode is expecting `XCBBuildService`. You also pass in the Xcode version, so your logs will be split based on the Xcode version.
+
+The `XCBBuildService` lives here: `$(XCODE_DIR)/Contents/SharedFrameworks/XCBuild.framework/PlugIns/XCBBuildService.bundle/Contents/MacOS/XCBBuildService`
+
+`write_shim.sh` will create a fake `XCBBuildService` that simply calls the `BazelXCBBuildService` while redirecting `STDERR` to a log file in `/tmp/Bazel-trace/`.
+
+### Prepping for a new version of Xcode:
+* Create a new module inside `/Sources` following the established pattern. (e.g. `XCBProtocol_14_0`)
+* You'll notice we make use of symlinks to old versions. Rather than copying every single file into the new `XCBProtocol` version, only copy files you need to change. For the rest, create a symlink to the most recent version of the file. You'll see lots of symlinks all the way back to Xcode 11.4!
+* Update `Examples/BazelXCBBuildService/BUILD.bazel` and `Examples/BazelXCBBuildService/Packge.swift` to point at that new module
+* Update `Examples/BazelXCBBuildService/Sources/RequestHandler.swift` to assign the appropriate `RequestPayload` and `ResponsePayload` types to the `BazelXCBBuildServiceRequestPayload` and `BazelXCBBuildServiceResponsePayload` typealiases.
+* Rename the original `XCBBuildService` in the Xcode bundle to `XCBBuildService.original`, just in case you want it again.
+
+### Development loop:
+* Rebuild: `bazel build BazelXCBBuildService`
+* Move it into the proper folder: `My-New-Xcode.app/Contents/SharedFrameworks/XCBuild.framework/PlugIns/XCBBuildService.bundle/Contents/MacOS/BazelXCBBuildService`
+* Run `write_shim.sh` to set up logging into `/tmp/Bazel-trace/`
+* Restart Xcode
+* Do some actions in Xcode, then view the logs, are there any errors?
+* Make a change in the project fixing an error or adding extra logs you deem necessary.
+* Repeat
+
+## Xcode Debugger
+
+To debug `BazelXCBBuildService` while it is being used by Xcode, you will need two versions of Xcode. (e.g. 12.5.0 and 13.0.0)
```

Comment:
This isn't 100% correct. You can also launch another instance of the same Xcode, just without the proxy. You would use the command line, set the env variable `XCBBUILDSERVICE_PATH` to point to the backed up version, and launch the binary. The Target Makefile should have an example of this.

Reply:
I don't fully understand your workflow here. I'll be opening a PR shortly with the changes you've suggested here. Could you elaborate there some more?

Reply:
Xcode looks to see if `XCBBUILDSERVICE_PATH` is set. If it is, it will use that instead of `Contents/SharedFrameworks/XCBuild.framework/PlugIns/XCBBuildService.bundle/Contents/MacOS/XCBBuildService`. So you can have the "main" Xcode launch, even if it has a proxy installed, by calling `env XCBBUILDSERVICE_PATH=/Application/Xcode.app/Contents/SharedFrameworks/XCBuild.framework/PlugIns/XCBBuildService.bundle/Contents/MacOS/XCBBuildService.original /Application/Xcode.app/Contents/MacOS/Xcode`. This will launch a new instance of that Xcode, which uses the original XCBBuildService.

Reply:
Ah cool, OK. I'm going to copy/paste some of that into the docs. :)